com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CheckForOperationsInProgressStep.java

```diff
@@ -0,0 +1,130 @@
+                                                           List<CloudServiceExtended> servicesToProcess) {
+        return servicesToProcess.parallelStream()
+                                .map(service -> getExistingService(controllerClient, service))
+                                .filter(Objects::nonNull)
+                                .collect(Collectors.toList());
+    }
+
+    private CloudServiceExtended getExistingService(CloudControllerClient controllerClient, CloudServiceExtended service) {
+        CloudService existingService = controllerClient.getService(service.getName(), false);
+        if (existingService != null) {
+            return ImmutableCloudServiceExtended.builder()
+                                                .from(service)
+                                                .metadata(existingService.getMetadata())
+                                                .build();
+        }
+        return null;
+    }
+
+    private Map<CloudServiceExtended, ServiceOperation> getServicesInProgressState(ExecutionWrapper execution,
+                                                                                   List<CloudServiceExtended> existingServices) {
+        Map<CloudServiceExtended, ServiceOperation> servicesOperation = new HashMap<>();
+        for (CloudServiceExtended existingService : existingServices) {
+            ServiceOperation lastServiceOperation = serviceOperationGetter.getLastServiceOperation(execution, existingService);
+            ServiceOperation serviceOperationInProgress = getServiceOperationInProgress(lastServiceOperation);
+            if (serviceOperationInProgress != null) {
+                servicesOperation.put(existingService, serviceOperationInProgress);
+            }
+        }
+        return servicesOperation;
+    }
+
+    private ServiceOperation getServiceOperationInProgress(ServiceOperation lastServiceOperation) {
+        if (lastServiceOperation != null && lastServiceOperation.getState() == ServiceOperationState.IN_PROGRESS) {
+            return lastServiceOperation;
+        }
+        return null;
+    }
+
+    private Map<String, ServiceOperationType>
+            getServicesOperationTypes(Map<CloudServiceExtended, ServiceOperation> servicesInProgressState) {
+        return servicesInProgressState.entrySet()
+                                      .stream()
+                                      .collect(Collectors.toMap(serviceName -> serviceName.getKey()
+                                                                                          .getName(),
+                                                                serviceOperationType -> serviceOperationType.getValue()
+                                                                                                            .getType()));
+    }
+
+    private List<CloudServiceExtended> getServicesData(Map<CloudServiceExtended, ServiceOperation> servicesInProgressState) {
```

Comment:
What is services data? Rename it to something more meaningful.
Also, rename the variable names, as well

Reply:
done